### ISSUE TITLE
TryOpen was removed?

### ISSUE THREAD

**Issue by egorpugin** [GiovanniDicanio/WinReg#31]
Hi,

Seems like it is.
But how do we open keys now without exception?
The code is
```
winreg::RegKey kits;
auto r = kits.TryOpen(HKEY_LOCAL_MACHINE, root, access);
if (r.IsOk())
{
```

**Comment by GiovanniDicanio** [comment]
Hi, 

I fixed that with pull request #32 .

However, I don't believe that basically adding Try methods for every existing method (almost 2X bloating the public interface) is a good thing. I designed this library with C++ _exceptions_ in mind as the main way to signal error conditions of the underlying Windows C API calls.

Thanks.

**Comment by egorpugin** [comment]
Thank you!

> I don't believe that basically adding Try methods for every existing method

also

> I designed this library with C++ exceptions in mind

I'm not sure about every method too, but trying to iterate some set of possible keys and checking what exists is quite common operation. This will be a bit costly with exceptions.
